keps/sig-api-machinery/20190322-server-side-get-to-ga.md

```diff
@@ -0,0 +1,119 @@
+---
+title: Graduate Server-side Get and Partial Objects to GA
+authors:
+  - "@smarterclayton"
+owning-sig: sig-api-machinery
+participating-sigs:
+  - sig-cli
+reviewers:
+  - "@lavalamp"
+  - "@soltysh"
+  - "@liggitt"
+approvers:
+  - "@liggitt"
+  - "@pwittrock"
+editor: TBD
+creation-date: 2019-03-22
+last-updated: 2019-03-22
+status: provisional
+see-also:
+  - "https://github.com/kubernetes/community/blob/master/contributors/design-proposals/api-machinery/server-get.md"
+replaces:
+superseded-by:
+---
+
+# Graduate Server-side Get and Partial Objects to GA
+
+## Table of Contents
+
+- [Title](#title)
+  - [Table of Contents](#table-of-contents)
+  - [Release Signoff Checklist](#release-signoff-checklist)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+    - [1.15](#1-15)
+    - [1.16](#1-16)
+    - [1.17](#1-17)
+    - [Implementation Details](#implementation-details)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Version Skew Strategy](#version-skew-strategy)
+  - [Implementation History](#implementation-history)
+
+## Summary
+
+Server-side columnar formatting and partial object metadata has been in beta since Kube 1.10 and as of 1.15 is consistently implemented and in wide use as part of `kubectl` and other web interfaces. This document outline required steps to graduate it to GA. 
+
+## Motivation
+
+The user of server-side printing with CRDs and Aggregated APIs is common and is a key part of providing equivalence between built-in and extension APIs for administrator usability. We have not needed to update the schema since 1.10, proving that it is ready to ship. Promoting it to GA in 1.15 will allow us to declare the kubectl portion feature complete and remove the legacy printers, and also update controllers that would benefit from use of PartialObjectMetadata (garbage collector, namespace controller, and quota counter) without fear of deprecation. PartialObjectMetadata allows these controllers to perform protobuf list actions efficiently when retrieving only the core object metadata.
+
+### Goals
+
+Server-side printing has no outstanding feature requests now that full WATCH support has been implemented in 1.15. It is ready to move to GA by promoting the resources.
```

Comment:
@smarterclayton - you're talking about this one: https://github.com/kubernetes/kubernetes/pull/71548, right?

Reply:
Yes, thank you will link.